portfolio/src/main/java/com/google/sps/servlets/Comment.java

```diff
@@ -0,0 +1,80 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.Entity;
+
+public class Comment {
+    private final String text;
+    private final String user;
+    private final long timestamp;
+    private int upvotes;
+    private int downvotes;
+    private long key;
+
+    // Constructor to use when comment is created
+    public Comment(String text, String user, long timestamp){
+        this.text = text;
+        this.user = user;
+        this.timestamp = timestamp;
+        this.upvotes = 0;
+        this.downvotes = 0;
+    }
+
+    // Constructor to use when comment is imported from Datastore
+    public Comment(Entity entity){
+        this.key = entity.getKey().getId();
+        this.text = (String)entity.getProperty("text");
+        this.user = (String)entity.getProperty("user");
+        this.timestamp = (long)entity.getProperty("timestamp");
+        this.upvotes = (Integer)entity.getProperty("upvotes");
+        this.downvotes = (Integer)entity.getProperty("downvotes");
+    }
+
+    public Comment(String text, String user, long timestamp, int upvotes, int downvotes, long key) {
+        this.text = text;
+        this.user = user;
+        this.timestamp = timestamp;
+        this.upvotes = upvotes;
+        this.downvotes = downvotes;
+        this.key = key;
+    }
```

Comment:
A good practice in Java is to have one "primary constructor" which is the only constructor that assigns fields. All other constructors "telescope" to the designated constructor, which makes refactoring the class much easier as there is only one place that the fields get assigned instead of three. Here is a [short example](https://www.yegor256.com/2015/05/28/one-primary-constructor.html) -- hope that helps illustrate the idea.

Reply:
Oh okay, its like calling super() when you inherit (kind of). Let me get on that